cli/src/customize/mod.rs

```diff
@@ -0,0 +1,362 @@
+//! Customize mode parser and evaluation logic.
+//!
+//! The customize mode is a way to interact with a specific Nickel program on the command line by
+//! setting or overriding values directly from the command line. It's currently enabled by the
+//! freeform marker `--`.
+use std::{collections::HashMap, ffi::OsString};
+
+use clap::Parser;
+use nickel_lang_core::{
+    combine::Combine,
+    eval::cache::lazy::CBNCache,
+    identifier::LocIdent,
+    program::{FieldPath, Program},
+    repl::query_print,
+    term::{
+        record::{Field, RecordData},
+        LabeledType, MergePriority, RuntimeContract, Term,
+    },
+    typ::{RecordRowF, RecordRowsIteratorItem, Type, TypeF},
+};
+
+use crate::error::CliResult;
+
+pub mod interface;
+
+use interface::{FieldInterface, TermInterface};
+
+/// The value name used through the CLI to indicate that an option accepts any Nickel expression as
+/// a value.
+const ASSIGNMENT_SYNTAX: &str = "FIELD_PATH=NICKEL_EXPRESSION";
+
+const EXPERIMENTAL_MSG: &str =
+    "[WARNING] Customize mode is experimental. Its interface is subject to breaking changes.";
+
+#[derive(clap::Parser, Debug)]
+pub struct CustomizeMode {
+    /// \[WARNING\] Customize mode is experimental. Its interface is subject to breaking changes.
```

Comment:
No, but we have to escape them, because otherwise Rustdoc's markdown parsers complains that it's a dangling link. However, it's fixed incidentally by #1712, which makes a pass on the interface of the customize mode.